test/integration/provider/sessions_controller_test.rb

```diff
@@ -37,4 +37,21 @@ class Provider::SessionsControllerTest < ActionDispatch::IntegrationTest
 
     assert_redirected_to "http://example.net/?provider_id=#{account.id}&user_id=#{account.admin_user.id}"
   end
+
+  test "redirect users to SSO when there's only one authentication provider and enforce_sso is on" do
```

Comment:
This is good 👍 but it lacks one scenario for testing: when there is more than one authentication provider but for non-published ones 😄